pkg/cosign/fulcioverifier/ctl/verify.go

```diff
@@ -27,20 +27,43 @@ import (
 	ct "github.com/google/certificate-transparency-go"
 	ctx509 "github.com/google/certificate-transparency-go/x509"
 	"github.com/google/certificate-transparency-go/x509util"
-	"github.com/sigstore/cosign/v2/cmd/cosign/cli/fulcio/fulcioverifier/ctutil"
 	"github.com/sigstore/cosign/v2/pkg/cosign/env"
+	"github.com/sigstore/cosign/v2/pkg/cosign/fulcioverifier/ctutil"
 
 	"github.com/sigstore/sigstore/pkg/cryptoutils"
 	"github.com/sigstore/sigstore/pkg/tuf"
 )
 
-// This is the CT log public key target name
-var ctPublicKeyStr = `ctfe.pub`
+// LogIDMetadata holds information for mapping a key ID hash (log ID) to associated data.
+type LogIDMetadata struct {
+	PubKey crypto.PublicKey
+	Status tuf.StatusKind
+}
+
+// This is the map of CTLog Public Keys indexed by log ID that's used in
+// verification of SCTs.
+type TrustedCTLogPubKeys struct {
+	// A map of Public Keys indexed by log ID
+	Keys map[string]LogIDMetadata
+}
+
+func NewTrustedCTLogPubKeys() TrustedCTLogPubKeys {
+	return TrustedCTLogPubKeys{Keys: make(map[string]LogIDMetadata, 0)}
+}
 
-// logIDMetadata holds information for mapping a key ID hash (log ID) to associated data.
-type logIDMetadata struct {
-	pubKey crypto.PublicKey
-	status tuf.StatusKind
+// AddCTLogPubKey adds a public key with the proper index (constructed) from the
+// Public Key representing the PEM-encoded Rekor key and a status.
+func (t *TrustedCTLogPubKeys) AddCTLogPubKey(pemBytes []byte, status tuf.StatusKind) error {
+	pubKey, err := cryptoutils.UnmarshalPEMToPublicKey(pemBytes)
+	if err != nil {
+		return err
+	}
+	keyID, err := ctutil.GetCTLogID(pubKey)
```

Comment:
I think this might be the same alg as the rekor log ID function too

Reply:
Yes, ideally we'd get rid of all the snowflake things from the Rekor because I think it's clearer they are generic CT related methods and keep all the things that are Rekor specific contained in the Rekor packages.